lib/ClangImporter/ImportDecl.cpp

```diff
@@ -5636,6 +5660,77 @@ Decl *SwiftDeclConverter::importGlobalAsInitializer(
   return result;
 }
 
+Decl *SwiftDeclConverter::importCXXMethodDecl(
+    const clang::CXXMethodDecl *decl, DeclName name, DeclContext *dc,
+    Optional<ImportedName> correctSwiftName) {
+  auto isStatic = decl->isStatic();
+  auto loc = Impl.importSourceLoc(decl->getLocation());
+  if (dc->getSelfProtocolDecl() && isStatic) {
+    Impl.SwiftContext.Diags.diagnose(loc, diag::swift_name_protocol_static,
+                                     /*isInit=*/false);
+    Impl.SwiftContext.Diags.diagnose(loc, diag::note_while_importing,
+                                     decl->getName());
+    return nullptr;
+  }
```

Comment:
I can't imagine what situation would cause us to import a C++ class as a protocol.